Sources/RealModule/AugmentedArithmetic.swift

```diff
@@ -9,18 +9,85 @@
 //
 //===----------------------------------------------------------------------===//
 
+/// A namespace for "augmented arithmetic" operations for types conforming to
+/// `Real`.
+///
+/// Augmented arithmetic refers to a family of algorithms that represent
+/// the results of floating-point computations using multiple values such that
+/// either the error is minimized or the result is exact.
 public enum Augmented { }
 
 extension Augmented {
+  /// The product `a * b` represented as an implicit sum `head + tail`.
+  ///
+  /// `head` is the correctly rounded value of `a*b`. If no overflow or
+  /// underflow occurs, `tail` represents the rounding error incurred in
+  /// computing `head`, such that the exact product is the sum of `head`
+  /// and `tail` computed without rounding.
+  ///
+  /// This operation is sometimes called "twoProd" or "twoProduct".
+  ///
+  /// Edge Cases:
+  /// -
+  /// - `head` is always the IEEE 754 product `a * b`.
+  /// - If `head` is not finite, `tail` is unspecified and should not be
+  ///   interpreted as having any meaning (it may be `NaN` or `infinity`).
+  /// - When `head` is close to the underflow boundary, the rounding error
+  ///   may not be representable due to underflow, and `tail` will be rounded.
+  ///   If `head` is very small, `tail` may even be zero, even though the
+  ///   product is not exact.
+  /// - If `head` is zero, `tail` is also a zero with unspecified sign.
+  ///
+  /// Postconditions:
+  /// -
+  /// - If `head` is normal, then `abs(tail) < abs(head.ulp)`.
+  ///   Assuming IEEE 754 default rounding, `abs(tail) <= abs(head.ulp)/2`.
```

Comment:
Are the “abs” on the right-hand sides necessary? I was under the impression that `ulp` was guaranteed to be non-negative for normal values. If so, the same thing applies to the doc comment for `sum(large:small:)`.

Reply:
That's right.